lua/entities/gmod_wire_expression2/core/entity.lua

```diff
@@ -89,6 +89,11 @@ e2function number entity:id()
 	return this:EntIndex()
 end
 
+e2function number entity:creationID()
+	if not IsValid(this) then return 0 end
```

Comment:
I suggest to cut it short (makes it easier to read):
`return IsValid(this) and this:GetCreationID() or 0`